includes/class-newspack-popups-segmentation.php

```diff
@@ -181,5 +184,37 @@ function( $cat ) {
 	public static function get_segmentation_endpoint() {
 		return plugins_url( '../api/segmentation/index.php', __FILE__ );
 	}
+
+	/**
+	 * Create the clients and events tables.
+	 */
+	public static function create_database_table() {
+		global $wpdb;
+		$events_table_name = Segmentation::get_events_table_name();
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
+		if ( $wpdb->get_var( $wpdb->prepare( 'SHOW TABLES LIKE %s', $events_table_name ) ) != $events_table_name ) {
+			$charset_collate = $wpdb->get_charset_collate();
+
+			$sql = "CREATE TABLE $events_table_name (
+				id bigint(20) NOT NULL AUTO_INCREMENT,
+				created_at date NOT NULL,
+				-- type of event
+				type varchar(20) NOT NULL,
+				-- Unique id of a device/browser pair
+				client_id varchar(100) NOT NULL,
+				-- Article ID
+				post_id bigint(20),
+				-- Article categories IDs
+				category_ids varchar(100),
+				UNIQUE KEY client_id_post_id (client_id, post_id),
```

Comment:
Won't this table also conceivably be queried based on `client_id` alone, `type` alone, and even `post_id` alone? If so, we may need more indexes.

Reply:
This should do the trick: 7ac8d413e03537057fef2094a05452d95d9eaad6